### PR TITLE
Mdct 2630 - allow approver and project officer roles to uncertify programs

### PR DESCRIPTION
**Pull request by britt-mo**
### Description
`This PR creates a new app role called 'CMS_APPROVER' to allow for the idm roles 'APPROVER' and 'PROJECT_OFFICER' to uncertify reports along with 'BUSINESS_OWNER_REP'.  

### Related ticket(s)

MDCT-2630

---
### How to test
Login as an admin and generate reports. If no certified and submitted reports exist, login as a state user and submit an existing report. Then login as the users provided below and confirm that each user sees the `Uncertify` option on submitted programs and can successfully uncertify them. You can login as the state user for the corresponding state for the uncertified program to see that it is back to `in progress` and the user can edit and resubmit.

cms_user - business owner rep
`cms.user@test.com`

cms_admin - project officer
`cms.admin@tests.com`

cms_approver - approver
`help.approver@test.com`

### Important updates

---
### Author checklist

- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://bit.ly/3zPrxuZ) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

**Review by ajaitasaini — DISMISSED** [review]
tested + lgtm! one question though - what's the difference between a cms_approver and cms_user?

File: services/app-api/libs/authorization.ts [hunk 1]
```diff
@@ -140,6 +140,7 @@ export const getUserCredentialsFromJwt = (event: APIGatewayProxyEvent) => {
 export const mapIdmRoleToAppRole = (idmRole: IdmRoles) => {
   switch (idmRole) {
     case IdmRoles.APPROVER:
+      return AppRoles.CMS_APPROVER;
     case IdmRoles.HELP:
       return AppRoles.HELP_DESK;
```
Comment (benmartin-coforma) [inline services/app-api/libs/authorization.ts:145]: Note: If there were any permission that AppRoles.HELP_DESK already had, which AppRoles.CMS_APPROVER was not also given in this PR, then this change would effectively be stripping permissions from IdmRoles.APPROVER. But, Britt was very thorough: there is no such permission.

In other words, there are no unintended side effects here.

**Review by benmartin-coforma — COMMENTED** [review]
This looks really good! I've got one small change request before approving.

File: services/app-api/handlers/stateStatus/update.ts [hunk 2]
```diff
@@ -43,9 +43,13 @@ export const updateStateStatus = handler(async (event, _context) => {
     };
 
     await dynamoDb.update(updateStateStatusparams);
-  } else if (user.role === AppRoles.CMS_USER) {
+  } else if (
+    user.role === AppRoles.CMS_USER ||
+    AppRoles.CMS_ADMIN ||
+    AppRoles.CMS_APPROVER
```
[the remark this answers is not in the document]
Comment (britt-mo) [reply services/app-api/handlers/stateStatus/update.ts:49]: will fix, thanks!

**Comment by britt-mo** [comment]
> tested + lgtm! one question though - what's the difference between a cms_approver and cms_user?



> tested + lgtm! one question though - what's the difference between a cms_approver and cms_user?

cms_user is a role that is only associated with business_owner_rep, and I was informed that the role could not be shared with other idmRoles, so cms_approver was created to give approver and project_officer the ability to uncertify.

**Review by benmartin-coforma — APPROVED** [review]
Looks great!

**Review by ajaitasaini — APPROVED** [review]
great job!! and ty for the clarification

**Pull request merged**